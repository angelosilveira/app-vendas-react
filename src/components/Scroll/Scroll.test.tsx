import '@testing-library/jest-dom';
import { renderWithTheme } from 'src/utils/tests/renderWithTheme';
import { it, describe } from 'vitest';
import { Scroll } from '.';

describe('<Scroll />', () => {
  it('should render the loading', () => {
    const { container, getByTestId } = renderWithTheme(
      <Scroll data-testid="scroll">
        <span>App vendas</span>
      </Scroll>
    );

    expect(getByTestId('scroll')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
