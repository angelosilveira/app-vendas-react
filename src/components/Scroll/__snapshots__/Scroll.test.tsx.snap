// Vitest Snapshot v1

exports[`<Scroll /> > should render the loading 1`] = `
.c0 {
  position: relative;
  height: calc(100vh - 215px);
  margin-right: -15px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 .cover-bar {
  position: absolute;
  background: #f4f4f4;
  height: 100%;
  top: 0;
  right: 0;
  width: 0.4em;
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
  opacity: 1;
  pointer-events: none;
}

.c0:hover .cover-bar {
  opacity: 0;
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
}

.c1 {
  position: absolute;
  width: 100%;
  left: 0;
  overflow-y: scroll;
  height: 100%;
  padding-right: 10px;
}

.c1::-webkit-scrollbar {
  width: 0.4em;
}

.c1::-webkit-scrollbar,
.c1::-webkit-scrollbar-thumb {
  overflow: visible;
  border-radius: 4px;
}

.c1::-webkit-scrollbar-thumb {
  background: rgba(0,0,0,0.2);
}

<div
  class="c0"
  data-testid="scroll"
>
  <div
    class="c1"
  >
    <span>
      App vendas
    </span>
  </div>
  <div
    class="cover-bar"
  />
</div>
`;
